### ISSUE TITLE
reference tag and name

### ISSUE THREAD

**Issue by hainm** [Amber-MD/cpptraj#100]
ack,

I've just realized that `[REF]` is not a name. I need to explicitly use ``[ ]`` to tag. Now I see why @drroe mentioned `tag` and `name` before.

Currently, I can not specify reference frame by name, right?

For example, this one should work (cpptraj man)
```bash
trajin Test1.crd
reference 1LE1.NoWater.Xray.rst7 [xray]
reference Test1.crd lastframe [last]
reference Test2.crd 225 [open]
rms Xray ref [xray] :2-12@CA out rmsd.dat
rms Last ref [last] :2-12@CA out rmsd.dat
rms Open ref [open] :2-12@CA out rmsd.dat
```

but this one does not, right? (I've tried and got error reference not found)
```
trajin Test1.crd
reference 1LE1.NoWater.Xray.rst7 xray
reference Test1.crd lastframe last
reference Test2.crd 225 open
rms Xray ref xray :2-12@CA out rmsd.dat
rms Last ref last :2-12@CA out rmsd.dat
rms Open ref open :2-12@CA out rmsd.dat
```

this is the reason why I was stuck in this issue
https://github.com/Amber-MD/cpptraj/issues/35

**Comment by hainm** [comment]
ideally, I wish we can add `name` 

trajin Test1.crd
reference 1LE1.NoWater.Xray.rst7 ``name xray``
reference Test1.crd lastframe ``name last``
reference Test2.crd 225 ``name open``
rms Xray ref ``xray`` :2-12@CA out rmsd.dat
rms Last ref ``last`` :2-12@CA out rmsd.dat
rms Open ref ``open`` :2-12@CA out rmsd.dat

**Comment by hainm** [comment]
oh, I just try to make a name for reference in Action_NativeContacts and refer its name when doing the action. It works. 

But I still want to hear your over idea when you have time.

**Comment by drroe** [comment]
The "tag" functionality is from a time when topologies and reference structures were stored in their own separate arrays. It's still supported for backwards compatibility. I'll consider adding the `name` keyword to `reference` and `parm` - it's not a bad idea.

**Comment by hainm** [comment]
> I'll consider adding the name keyword to reference and parm - it's not a bad idea.

I think this will be very convenient for pytraj. pytraj can take the reference from DSL. See my desired command below.

```python
        traj = pt.iterload("./data/md1_prod.Tc5b.x", "./data/Tc5b.top")

        commands = ['rmsd @CA',
                    'distance :3 :7',
                    'distance     :3 :7',
                    'vector :2 :3',
                    'rmsd ref my_ref',] # my_ref is the name DataSet_Coords_REF

        dslist = CpptrajDatasetList()
        actlist = ActionList(commands, traj.top, dslist=dslist)

        # add reference for above command with name 'my_ref`
        dslist.add_set('ref_frame', 'my_ref')

        for frame in traj:
            actlist.do_actions(frame)
```

**Comment by hainm** [comment]
And I realize that for Action, we don't need to write down the word `name` but for Analysis, we can.

What's your opinion about adding `name` for Dataset in Action too?

```bash
rms first name my_rmsd
```
which is equal to current implementation
```bash
rms my_rmsd first
```

**Comment by drroe** [comment]
Addressed by #245
